src/include/argdata.h

```diff
@@ -34,6 +34,12 @@ typedef __argdata_t argdata_t;
 #define _ARGDATA_T_DECLARED
 #endif
 
+typedef struct {
+	__size_t index;
```

Comment:
I'm not convinced that it's worth having this functionality part of the public API. Especially now that we can iterate over maps and sequences, it becomes trivial for the caller to keep track of the index, right? For the callback-based iteration this would have made more sense.